### PR TITLE
add requirements for gui and add dockerfile

### PR DESCRIPTION
**Pull request by Bacilika**
This PR will not work for docker yet and is a start for docker support. 
To run the web gui. run ./start_ossqa_web.sh from root. There will not be any result to display but requirement selection is working.

### CODE REVIEW

File: src/frontend/OSSQAWebAPI/OSSQAWebAPI/Components/Pages/Home.razor
Comment (edvgi966) [inline src/frontend/OSSQAWebAPI/OSSQAWebAPI/Components/Pages/Home.razor:1]: Some docstrings and comments would be greatly appreciated

**Review by edvgi966 — APPROVED** [review]
Minor comment on missing docstrings in the C# files

**Comment by grufe** [comment]
Added proposed fix to branch "web-api-fixes". 
These fixes to the Dockerfiles and web_api.py should fix the response for the website.

These could be pushed into this branch for further development.

**Review by parslie — COMMENTED** [review]
I've looked at the code and can't find anything wrong, but all I get when I click the analyze button is the number "1" and a spinning circle. Others seem to get different results, so I will try running it on my desktop when I get home.

**Review by grufe — APPROVED** [review]
Looks good!

**Review by KarlssonPaGolvet — APPROVED** [review]
Everythong looks good to me

**Pull request merged**